lesson07-Cakephp　課題②実装

ayktmr opened the pull request:
CakePHP課題②が完了しました。
レビューの程、宜しくお願い致します。

・商品詳細情報欄の追加
・出品画像の追加（アップロード、詳細画面での表示）
・終了時刻カウントダウンタイマー追加

ayktmr committed:
add カラム追加、入力フォーム追加

ayktmr committed:
add 商品画像の保存先フォルダを追加

ayktmr committed:
add goods_imagesフォルダ内画像を ignore設定

ayktmr committed:
add ファイルアップロード時の値がうまく渡せていないソース

ayktmr committed:
error: Cannot validate mimetype for a missing file

ayktmr committed:
fix ファイルアップロード・バリデーションを修正

ayktmr committed:
add 終了日迄のカウントダウンタイマー機能を追加

ayktmr committed:
fix オークション終了時のコメント不具合を調整

ayktmr committed:
add カウントダウンタイマーを赤字に変更

ayktmr committed:
fix タイムゾーン修正

ayktmr committed:
add エラー表示の見た目を追加

choimake commented on src/Model/Table/BiditemsTable.php:
```diff
@@ -70,12 +70,18 @@ public function validationDefault(Validator $validator)
-            ->notEmptyString('name');
+            ->notEmpty('name', '※商品名を入力して下さい。');
+
+        $validator
+            ->scalar('goods_detail')
+            ->maxLength('goods_detail', 10000)
+            ->requirePresence('goods_detail', 'create')
+            ->notEmpty('goods_detail', '※商品詳細を入力して下さい。');
```
こちらにカラムの追加が実装されていますが、このカラムを追加するためのDDL文に関する情報がなくこちらの環境では動作確認がとれません。
Migrationの仕組みを利用して、不足しているカラムを追加する処理を追加してください。

ayktmr replied:
ご指摘ありがとうございます。
知らない機能を知ることが出来ました！
ご確認の程、宜しくお願い致します。

choimake reviewed (commented):
コメントしました。

まずコミットのコメントに関してですが、一通りのことに関しては書かれているようなので良いと思います。
「add」「fix」みたいなprefixが書かれていると、「何をしたのか」がすごくわかりやすいと感じます。

もし改善点があるとしたら「何故その処理を実装したのか」という情報があるとより良くなると思います。（特に修正時）

例えば「オークション終了時のコメント不具合を調整」という内容をみると「どんなコメント不具合をどう調整したのか？」という情報があると、後々の人が「このコードの変更どういう理由でかけたんだろう？」と疑問に思った時に、gitのコメントでそれに答えていると、調査コストが大きくさがります。

コード変更の意図は実装した本人にしかわからない情報です。その意図を残すようにするともっと良いコメントになると思います。

これは余談で修正不要ですが、ブランチ名が「cakephp」なのはちょっと意味がわからないのでちょっといただけないです。
もう少し意図を示した名前をつけるようにしましょう。


あとマイグレーションに今回の実装で追加したであろうカラムの情報がないため、こちらで動作確認できませんでした。
こちら対応お願いします。

ayktmr committed:
add テーブル定義変更後、差分のマイグレーションファイルを作成

ayktmr commented:
> もし改善点があるとしたら「何故その処理を実装したのか」という情報があるとより良くなると思います。（特に修正時）
> 例えば「オークション終了時のコメント不具合を調整」という内容をみると「どんなコメント不具合をどう調整したのか？」という情報があると、後々の人が「このコードの変更どういう理由でかけたんだろう？」と疑問に思った時に、gitのコメントでそれに答えていると、調査コストが大きくさがります。

ご指摘ありがとうございます。
自分でも後々確認した際に、何を？と思うのに活かせてなかったです。
今後は、その時の自分しか分からない情報ではなく、変更等の意図を残します。

> これは余談で修正不要ですが、ブランチ名が「cakephp」なのはちょっと意味がわからないのでちょっといただけないです。
> もう少し意図を示した名前をつけるようにしましょう。（今回は修正不要）

本当です。自分でも意味が分かりません...もっと意識します。

ayktmr committed:
fix アップロード時の超過ファイルサイズ数を修正

filesize 超過時のコメント誤りを変更：（10M→1M）

filesize確認時、単位をKBにしていた為、1M以下でも弾かれていた。
KB → B へ修正。

ayktmr committed:
add 終了日時を現在時刻から変更せず登録可から、日時指定する旨メッセージを表示するよう変更

ayktmr committed:
fix XMLHttpRequest非推奨に設定を変更

ayktmr committed:
fix カウントダウン中の「終了まで」コメントを終了直後に表示させないよう修正

ayktmr committed:
add viewアクションの実行を追加

view表示中はカウントダウンが「０」になっても finished の値は変更されない為、
カウントダウン進行中から０になった際には、viewアクションを実行させるように変更した

choimake commented on webroot/js/auction.js:
```diff
@@ -0,0 +1,63 @@
+// ▼ カウントダウンタイマーの設定
+function CountdownTimer(elm, tl, mes) {
+    this.initialize.apply(this, arguments);
+  }
+  CountdownTimer.prototype = {
+    initialize: function (elm, tl, mes, resend) {
```
この辺から以下、インデントがずれています。修正お願いします。

ayktmr replied:
きちんと確認せず失礼致しました。
修正しました。

choimake commented on webroot/js/auction.js:
```diff
@@ -0,0 +1,63 @@
+          me.countDown();
+        }, 10);
+      } else if((this.tl - today) <= 0) {
+        resend = window.location.href = "/mycakeapp/Auction/view/" + id; 
+        this.elem.innerHTML = this.mes;
```
~~（対応不要）
ここはfinishedのDOM要素取得して中のhtml書き換える対応でも良さそうですね。~~

choimake replied:
あ、リダイレクトしないとDBのfinishedの値変わらない仕様でしたっけ。これスルーしといてください。

ayktmr replied:
承知しました。
ありがとうございます。
DOMについて調べる事ができました。

choimake reviewed (commented):
確認しました。
全体的に丁寧な挙動といいよくできています。
またgitのコメントも実装の理由が記載されており、コードを読む上で大変助かりました。

ほぼOKなのですが、少しだけ修正して欲しい点があるので、そちら対応お願いします。

ayktmr committed:
style インデントの修正

choimake commented on webroot/js/auction.js:
```diff
@@ -0,0 +1,65 @@
+// ▼ カウントダウンタイマーの設定
+function CountdownTimer(elm, tl, mes) {
+    this.initialize.apply(this, arguments);
+    }
```
ここ、インデントひとつ深くないでしょうか？

ayktmr replied:
多いですね・・１番上で油断していたのかもしれません・・

ayktmr replied:
度々申し訳ありません。
他部分も合わせて修正致しました。

choimake reviewed (commented):
コメントしましたので確認お願いします。

ayktmr committed:
style インデントの再修正

choimake reviewed (approved):
対応ありがとうございました。LGTMです。
